### PR TITLE
fix issue: refactor MySQLJdbcQueryPropertiesExtension.

### LINKED ISSUE #21701: Refactor MySQLJdbcQueryPropertiesExtension
Improvement:

1, Refactor `mysqlConnectorVersion` as `final` and also rename to uppercase variable name `MYSQL_CONNECTOR_VERSION`; Refactor static code block as static method, assign the result to `MYSQL_CONNECTOR_VERSION`;

2, Extract a method to get `zeroDateTimeBehavior`;

### PR DESCRIPTION
**Pull request by pandaapo**
Fixes #21701.

---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [ ] I have passed maven check locally : `mvn clean install -B -T2C -DskipTests -Dmaven.javadoc.skip=true -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.

### CODE REVIEW

**Comment by pandaapo** [comment]
@sandynz Could you review this PR?

File: kernel/data-pipeline/dialect/mysql/src/main/java/org/apache/shardingsphere/data/pipeline/mysql/datasource/MySQLJdbcQueryPropertiesExtension.java [hunk 1]
```diff
@@ -47,14 +49,19 @@ public MySQLJdbcQueryPropertiesExtension() {
         queryProps.setProperty("useSSL", Boolean.FALSE.toString());
         queryProps.setProperty("rewriteBatchedStatements", Boolean.TRUE.toString());
         queryProps.setProperty("yearIsDateType", Boolean.FALSE.toString());
+        queryProps.setProperty("zeroDateTimeBehavior", getZeroDateTimeBehavior());
+        queryProps.setProperty("noDatetimeStringSync", Boolean.TRUE.toString());
+        queryProps.setProperty("jdbcCompliantTruncation", Boolean.FALSE.toString());
+
```
Comment (sandynz) [inline kernel/data-pipeline/dialect/mysql/src/main/java/org/apache/shardingsphere/data/pipeline/mysql/datasource/MySQLJdbcQueryPropertiesExtension.java:55]: This blank line could be deleted

File: kernel/data-pipeline/dialect/mysql/src/main/java/org/apache/shardingsphere/data/pipeline/mysql/datasource/MySQLJdbcQueryPropertiesExtension.java [hunk 2]
```diff
@@ -29,15 +29,17 @@
...
     
-    private static String mysqlConnectorVersion;
+    private static final String MYSQL_CONNECTOR_VERSION = initMysqlConnectorVersion();
     
-    static {
+    private static String initMysqlConnectorVersion() {
         try {
             Class<?> mysqlDriverClass = MySQLJdbcQueryPropertiesExtension.class.getClassLoader().loadClass("com.mysql.jdbc.Driver");
-            mysqlConnectorVersion = mysqlDriverClass.getPackage().getImplementationVersion();
+            String mysqlConnectorVersion = mysqlDriverClass.getPackage().getImplementationVersion();
```
Comment (sandynz) [inline kernel/data-pipeline/dialect/mysql/src/main/java/org/apache/shardingsphere/data/pipeline/mysql/datasource/MySQLJdbcQueryPropertiesExtension.java:37]: Variable name `mysqlConnectorVersion` should be `result`

**Review by sandynz — CHANGES_REQUESTED** [review]

**Comment by sandynz** [comment]
Hi @pandaapo , thanks for your contribution. I've just submit the review.

**Comment by pandaapo** [comment]
> Hi @pandaapo , thanks for your contribution. I've just submit the review.

Thanks for you review.

**Pull request merged**